modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/ScalaClientCodegen.java

```diff
@@ -275,28 +313,9 @@ public String toOperationId(String operationId) {
 
     @Override
     public String toVarName(String name) {
-        // sanitize name
-        name = sanitizeName(name); // FIXME: a parameter should not be assigned. Also declare the methods parameters as 'final'.
-
-        if("_".equals(name)) {
-          name = "_u";
-        }
-
-        // if it's all uppper case, do nothing
-        if (name.matches("^[A-Z_]*$")) {
-            return name;
-        }
-
-        // camelize (lower first character) the variable name
-        // pet_id => petId
-        name = camelize(name, true);
-
-        // for reserved word or word starting with number, append _
-        if (isReservedWord(name) || name.matches("^\\d.*")) {
```

Comment:
The logic to handle reserved words and word starting with number seems to be removed. Please take a look

For this change, please try to make it backward compatible with the current logic.